cmd/troubleshoot/cli/run.go

```diff
@@ -277,6 +313,142 @@ func loadSupportBundleSpecsFromURIs(ctx context.Context, kinds *loader.Troublesh
 	return nil
 }
 
+func validateSpecs(args []string, specYaml string) (*[]troubleshootv1beta2.SupportBundle, error) {
+	// Append redactor uris to the args
+	allArgs := append(args, viper.GetStringSlice("redactors")...)
+	kinds, err := specs.LoadFromCLIArgs(context.TODO(), nil, allArgs, viper.GetViper())
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if we have any collectors to run in the troubleshoot specs
+	if len(kinds.CollectorsV1Beta2) == 0 &&
+		len(kinds.HostCollectorsV1Beta2) == 0 &&
+		len(kinds.SupportBundlesV1Beta2) == 0 {
+		return nil, errors.New("no collectors specified to run")
+	}
+
+	for _, arg := range args {
+		err := checkSpecStructure(arg)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	for _, sb := range kinds.SupportBundlesV1Beta2 {
+		sb := sb
+		warning := validateTroubleshootSpecsItems(sb.Spec.Collectors, sb.Spec.Analyzers)
+		if warning != nil {
+			return nil, errors.New(warning.Warning())
+		}
+	}
+
+	// for _, c := range kinds.CollectorsV1Beta2 {
+	// 	mainBundle.Spec.Collectors = util.Append(mainBundle.Spec.Collectors, c.Spec.Collectors)
+	// }
+
+	// for _, hc := range kinds.HostCollectorsV1Beta2 {
+	// 	mainBundle.Spec.HostCollectors = util.Append(mainBundle.Spec.HostCollectors, hc.Spec.Collectors)
+	// }
+
+	return &kinds.SupportBundlesV1Beta2, nil
+}
+
+func validateTroubleshootSpecsItems(collectors []*troubleshootv1beta2.Collect, analyzers []*troubleshootv1beta2.Analyze) *types.ExitCodeWarning {
+	numberOfCollectors := len(collectors)
+	numberOfAnalyzers := len(analyzers)
+
+	if numberOfCollectors > 0 {
+		for _, c := range collectors {
+			if isStructEmpty(c) {
+				return types.NewExitCodeWarning("Wrong collector found")
+			}
+		}
+	} else {
+		return types.NewExitCodeWarning("No collectors found")
+	}
+
+	if numberOfAnalyzers > 0 {
+		for _, a := range analyzers {
+			if isStructEmpty(a) {
+				return types.NewExitCodeWarning("Wrong analyzer found")
+			}
+		}
+	} else {
+		return types.NewExitCodeWarning("No analyzers found")
+	}
+	return nil
+}
+
+func checkSpecStructure(path string) error {
+	if _, err := os.Stat(path); err == nil {
+		rawSpec, err := os.ReadFile(path)
+		if err != nil {
+			return types.NewExitCodeError(constants.EXIT_CODE_SPEC_ISSUES, err)
+		}
+
+		decoder := yaml.NewDecoder(strings.NewReader(string(rawSpec)))
+		var node yaml.Node
+		err = decoder.Decode(&node)
+		if err != nil {
+			return err
+		}
+
+		analyzerFields := listFieldNames(&troubleshootv1beta2.Analyze{})
+
+		analyzeFieldsInLowerCamelCase := strings.ToLower(strings.Join(analyzerFields, " "))
+
+		for _, n := range node.Content[0].Content { // Traverse the root map
```

Comment:
The idea you have with validating and reporting is correct, I'm however not sure we want to implement this feature by parsing yaml and inspecting fields like this. We'll be reinventing the wheel. There are a number of libraries that already exist that utilize yaml/json/openapi spec annotations. I've listed a few in https://github.com/replicatedhq/troubleshoot/issues/871 issue.

Reply:
great! I will have a look. In the meanwhile, I am checking the product team to see what their opinions. We can discuss the output of this command together.